compiler/AST/expr.cpp

```diff
@@ -6463,6 +6463,79 @@ CallExpr* ContextCallExpr::getRValueCall() {
 *                                                                           *
 ************************************* | ************************************/
 
+ForallExpr::ForallExpr(Expr* indices,
+                       Expr* iteratorExpr,
+                       Expr* expr,
+                       Expr* cond,
+                       bool maybeArrayType,
+                       bool zippered) :
+  Expr(E_ForallExpr),
+  indices(indices),
+  iteratorExpr(iteratorExpr),
+  expr(expr),
+  cond(cond),
+  maybeArrayType(maybeArrayType),
+  zippered(zippered)
+{
+  gForallExprs.add(this);
+}
+
+ForallExpr* ForallExpr::copyInner(SymbolMap* map) {
+  return new ForallExpr(
+    COPY_INT(indices),
+    COPY_INT(iteratorExpr),
+    COPY_INT(expr),
+    COPY_INT(cond),
+    maybeArrayType,
+    zippered);
+}
+
+void ForallExpr::replaceChild(Expr* old_ast, Expr* new_ast) {
+  if (old_ast == indices)
+    indices = new_ast;
+  else if (old_ast == iteratorExpr)
+    iteratorExpr = new_ast;
+  else if (old_ast == expr)
+    expr = new_ast;
+  else if (old_ast == cond)
+    cond = new_ast;
+  else
+    INT_FATAL(this, "unexpected case in ForallExpr::replaceChild");
+}
+
+void
+ForallExpr::verify() {
+  Expr::verify();
+  if (astTag != E_ForallExpr)
+    INT_FATAL(this, "bad ForallExpr::astTag");
+  INT_FATAL("ForallExprs should not persist past parser");
```

Comment:
This error message does not make sense to me. Why can't verify be called during parsing? Or accept, or getFirstChild, or getFirstExpr? I'd prefer if the message simply said that these functions were not implemented for ForallExpr.